### ISSUE TITLE
Updating modpack stuck at downloading mods 0%

### ISSUE THREAD

**Issue by Jordan231111** [killpowa/Twickt-Launcher#1231]
**Describe the bug**
Sometimes when updating modpacks, it becomes stuck at downloading mods 0 percent

**To Reproduce**
1. Go to update a modpack to a newer version (happened on multiple modpacks)

4. Stuck at 0 percent downloading mods

**Expected behavior**
For the Modpack to start updating and downloading the mods

**Possible solutions**
Some ideas of how you would solve this issue, if possible

**Screenshots**
![image](https://user-images.githubusercontent.com/79342877/147520472-a0331234-f124-4fe5-a65e-24d88e51e7ef.png)

**Operating System:**
- OS: [e.g. Windows]
- 10
- Java Version: 8 and 17
- Minecraft Version: 1.16.5 and 1.18.1
- Mod Loader: both forge and fabric

**Additional context**

**Comment by wmboeckman** [comment]
Could you provide a log?
![image](https://user-images.githubusercontent.com/47402550/147525802-64e745d9-12f3-4da8-94c3-f65a459ec4cb.png)
![image](https://user-images.githubusercontent.com/47402550/147526083-35838fe3-f851-4057-bfe9-0061f602caf5.png)

**Comment by CallMeSage** [comment]
I seem to be having the same issue as of now. Just updated AllTheMods 6 from 1.8.15 to 1.8.16.

https://i.imgur.com/RfVbyiy.png

**Comment by Jordan231111** [comment]
[-1641432365971.log](https://github.com/gorilla-devs/GDLauncher/files/7818754/-1641432365971.log)

**Comment by wmboeckman** [comment]
Both seem to be Issuers with the GDLauncher api throwing code 400 and 504, failing the download.

Some form of check should be made to be sure that when a file fails to download, it starts over.
And (if not already implemented) maybe even compare file checksums to make sure nothing got corrupted.

Best you can do for now is retry the update (maybe even backup the instance to be safe) a few times until it works?

**Comment by IvoryOwl92** [comment]
I've been having the exact same issue (400 and 504 error code).

Repairing the instance, deleting and starting over, closing GD Launcher and re-opening... none of it fixes the issue. I've tried to download two modpacks several times now. Nothing fixes it.

**Comment by NicTanghe** [comment]
I also get 400 / 504 when trying to dowload any mods (valhesia vanilla wnhansed)
```
Failed to load resource: the server responded with a status of 400 ()
instrument.ts:130 Error: Request failed with status code 400
    at t.exports (createError.js:16)
    at t.exports (settle.js:17)
    at XMLHttpRequest.d.onreadystatechange (xhr.js:62)
    at XMLHttpRequest.n (helpers.ts:88)
(anonymous) @ instrument.ts:130
addons-ecs.forgesvc.net/api/v2/addon/495693/file/3550299:1 Failed to load resource: the server responded with a status of 504 ()
createError.js:16 Uncaught (in promise) Error: Request failed with status code 504
    at t.exports (createError.js:16)
    at t.exports (settle.js:17)
    at XMLHttpRequest.d.onreadystatechange (xhr.js:62)
    at XMLHttpRequest.n (helpers.ts:88)
```

**Comment by cauanmendonca** [comment]
I'm having the same issue,on try download or repair the modpack Odyssey 2,some others modpacks have the same issues here but the lasted is Odyssey 2, i put the full log in the attachment
[-1642002972698.log](https://github.com/gorilla-devs/GDLauncher/files/7855986/-1642002972698.log)

**Comment by Alaric323** [comment]
This seems to be more of a Curseforge issue, sadly.  wonder if it's to do with the switch to Overwolf and the new mod policy...
![image](https://user-images.githubusercontent.com/4038592/149550871-1b132cee-238d-4fe4-bb5c-b4b6a34b7a74.png)

**Comment by Rhodderz** [comment]
Also getting this issue on multiple clients (ATLauncher as well), oddly MultiMC is fine.
I did notice going to the URLs failing via Curl does work but takes a good 5 to sometimes 10 seconds and wondering if it is a really small timeout set on the clients. I also could be paranoid but could also be CurseForge silently dropping GDLauncher and ATLauncher requests

**Comment by Jordan231111** [comment]
It happens very frequently too and takes many many tries to update a modpack :/

**Comment by Kooramah** [comment]
> This seems to be more of a Curseforge issue, sadly. wonder if it's to do with the switch to Overwolf and the new mod policy... ![image](https://user-images.githubusercontent.com/4038592/149550871-1b132cee-238d-4fe4-bb5c-b4b6a34b7a74.png)

This is exactly what I get from the console as well.

**Comment by hron84** [comment]
Same symptoms. 
Recommended workaround: provide a way to cancel an install/update (potentially rolling back the modpack to the previous state - if there's any - by doing update in temp folder and replacing folders later)
For 504 codes, if it appears, sleep a thread for a while, then re-try the download again. I'm pretty sure it could be a badly implemented throttling mechanism.

**Comment by MokahTGS** [comment]
Still getting this on the TNP 5 pack for 1.18.1

My Log: https://pastebin.com/4K5a15Nn

**Comment by koskenjuho** [comment]
Still not working. I really hope this is on the top priority right now. Friends can't join my server as they are not able to download ANY modpack atm. All packs get stuck to 0%

**Comment by Jordan231111** [comment]
I just use curseforge as an alternative at this point as it can take more than 10 attempts to redownload a modpack

**Comment by koskenjuho** [comment]
> I just use curseforge as an alternative at this point as it can take more than 10 attempts to redownload a modpack

Yeah, that wasn't even enough for my friends. +20 tries and not working. Used multimc to download and copied all the mods and configs to GD launcher folder and now it works.

**Comment by MokahTGS** [comment]
Yeah, I did the CurseForge route as well and copied my instances over...sheesh...please fix.

**Comment by thehairy** [comment]
I self-deployed the launcher from the master branch and I have absolutely no problems at all.

**Comment by Jaders2** [comment]
![image](https://user-images.githubusercontent.com/67982891/152399754-4fbfefb6-af48-4ac1-8fd3-fcfbcfba018f.png)

help

**Comment by Jordan231111** [comment]
> I self-deployed the launcher from the master branch and I have absolutely no problems at all.

looking at the recent commits it seems like they are fixing it hope a beta or something comes out soon..

**Comment by thehairy** [comment]
> > I self-deployed the launcher from the master branch and I have absolutely no problems at all.
> 
> looking at the recent commits it seems like they are fixing it hope a beta or something comes out soon..

You can download a build from the master branch [here at the bottom](https://github.com/gorilla-devs/GDLauncher/actions/runs/1766043701).

**Comment by Jordan231111** [comment]
Ok thank you this https://github.com/gorilla-devs/GDLauncher/suites/5107500519/artifacts/152971137
works well

**Comment by Jordan231111** [comment]
it was all working fine yesterday until this morning more 504 errors even in the latest master branch build lol
 https://paste-bin.xyz/36592 also the whole thing is not working even cant search mods
![image](https://user-images.githubusercontent.com/79342877/152642215-51806f6a-4aec-48fc-b399-71cb9b07bff2.png)

**Comment by blarfoon** [comment]
Our api server blew up, I’m gonna check what’s wrong in a few hours

**Comment by commandhat** [comment]
Had a similar error except it self-resolved... ironically, at the same time I opened this github page. Same 400 error in the logs.

**Comment by Midnex** [comment]
Anyone try hitting cancel when it says error can't connect on this message? I did, and it deleted all of my saved games and backups...

**Comment by jkwphysics** [comment]
Has anyone found a fix for this temporarily? I haven't had a single successful installation yet.  Been trying "multiple times"
![image](https://user-images.githubusercontent.com/51686467/155723425-f8300ed5-9b04-4e8e-bbf7-09dfcb1c5b27.png)

Edit: GDLauncher installed today with latest binary

**Comment by GooberMcNugget** [comment]
I am getting the same issue. Worked fine a couple days ago with the official release, then that stopped working so I downloaded the build linked in this thread, that hasn't worked for me... any updates on where this is at?

As a note in case it matters, I have tried both downloading directly within the launcher as well as downloading a zip from curseforge's website and then importing that into the launcher but neither have worked.
![image_2022-02-25_152021](https://user-images.githubusercontent.com/76923559/155804541-13667bc3-a52b-4611-9b21-f61d3bc770a7.png)

**Comment by AshtakaOOf** [comment]
@GooberMcNugget @jkwphysics use the gdlauncher beta for now because the beta use the new curseforge api

**Comment by GooberMcNugget** [comment]
@AshtakaOOf Thanks! At least the number gets past zero now!
